src/Stratis.Bitcoin.Features.Wallet/Interfaces/IWalletManager.cs

```diff
@@ -43,6 +43,13 @@ public interface IWalletManager
         /// <returns>A collection of spendable outputs</returns>
         IEnumerable<UnspentOutputReference> GetSpendableTransactionsInWalletForStaking(string walletName, int confirmations = 0);
 
+        /// <summary>
+        /// List all unspent transactions contained in a given wallet.
+        /// This is distinct from the list of spendable transactions. A transaction can be unspent but not yet spendable due to coinbase/stake maturity, for example.
+        /// </summary>
+        /// <returns>A collection of unspent outputs</returns>
+        IEnumerable<UnspentOutputReference> GetUnspentTransactionsInWallet(string walletName, int confirmations, Func<HdAccount, bool> accountFilter);
```

Comment:
Whats `accountFilter` for? I think this should actually be removed all together the account filter, that's the reason the api method to fetch accounts doesn't work for cold staking